src/aws-util/src/s3.rs

```diff
@@ -47,3 +48,36 @@ pub async fn list_bucket_path(
         })
         .transpose()?)
 }
+
+/// Upload an object to an S3 bucket. `mz_aws_util::s3_uploader::S3MultiPartUploader`
+/// should be used instead for large files.
+pub async fn upload_object<T: Into<ByteStream>>(
```

Comment:
Is these two worth being separate functions? It looks like it only saves the user from writing `.put_object()`/`.delete_object()` and `.send()`.